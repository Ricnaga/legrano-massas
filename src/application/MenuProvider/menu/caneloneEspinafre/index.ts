import { useReducer } from 'react';
import { UseMenuProviderData } from '../../hooks/useMenuProvider';
import { menu_caneloneEspinafre } from './items';

export enum CaneloneEspinafreActionsType {
  ADD_TO_CART = 'ADD_TO_CART',
  ADD_AMOUNT = 'ADD_AMOUNT',
  REMOVE_FROM_CART = 'REMOVE_FROM_CART',
}

type CaneloneEspinafreState = UseMenuProviderData;

type CaneloneEspinafreActions = {
  action: CaneloneEspinafreActionsType;
  payload: {
    id: string;
  };
};

const menuCaneloneEspinafre = (
  state: CaneloneEspinafreState,
  { action, payload }: CaneloneEspinafreActions,
) => {
  switch (action) {
    case CaneloneEspinafreActionsType.ADD_TO_CART: {
      const { items } = state;

      const selectItem = items.map((item) =>
        item.id === payload.id ? { ...item, selected: true, amount: 1 } : item,
      );
      return { ...state, items: selectItem };
    }
    case CaneloneEspinafreActionsType.ADD_AMOUNT: {
      const { items } = state;
      const addMountToItem = items.map((item) =>
        item.id === payload.id ? { ...item, amount: item.amount + 1 } : item,
      );

      return { ...state, items: addMountToItem };
    }

    case CaneloneEspinafreActionsType.REMOVE_FROM_CART: {
      const { items } = state;
      const removeAmountToItem = items.map((item) => {
        if (item.id === payload.id && item.amount - 1 > 0) {
          return { ...item, amount: item.amount - 1 };
        }
        if (item.id === payload.id && item.amount - 1 === 0) {
          return { ...item, amount: 0, selected: false };
        }

        return item;
      });

      return { ...state, items: removeAmountToItem };
    }
    default:
      return state;
  }
};

export const useCaneloneEspinafre = () => {
  const [caneloneEspinafre_state, dispatch] = useReducer(
    menuCaneloneEspinafre,
    menu_caneloneEspinafre,
  );

  return {
    data: { caneloneEspinafre_state },
    functions: { dispatchCaneloneEspinafre: dispatch },
  };
};
