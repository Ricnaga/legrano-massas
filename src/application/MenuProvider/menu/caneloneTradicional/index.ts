import { useReducer } from 'react';
import { UseMenuProviderData } from '../../hooks/useMenuProvider';
import { menu_caneloneTradicional } from './items';

export enum CaneloneTradicionalActionsType {
  ADD_TO_CART = 'ADD_TO_CART',
  ADD_AMOUNT = 'ADD_AMOUNT',
  REMOVE_FROM_CART = 'REMOVE_FROM_CART',
}

type CaneloneTradicionalState = UseMenuProviderData;

type CaneloneTradicionalActions = {
  action: CaneloneTradicionalActionsType;
  payload: {
    id: string;
  };
};

const menuCaneloneTradicional = (
  state: CaneloneTradicionalState,
  { action, payload }: CaneloneTradicionalActions,
) => {
  switch (action) {
    case CaneloneTradicionalActionsType.ADD_TO_CART: {
      const { items } = state;

      const selectItem = items.map((item) =>
        item.id === payload.id ? { ...item, selected: true, amount: 1 } : item,
      );
      return { ...state, items: selectItem };
    }
    case CaneloneTradicionalActionsType.ADD_AMOUNT: {
      const { items } = state;
      const addMountToItem = items.map((item) =>
        item.id === payload.id ? { ...item, amount: item.amount + 1 } : item,
      );

      return { ...state, items: addMountToItem };
    }

    case CaneloneTradicionalActionsType.REMOVE_FROM_CART: {
      const { items } = state;
      const removeAmountToItem = items.map((item) => {
        if (item.id === payload.id && item.amount - 1 > 0) {
          return { ...item, amount: item.amount - 1 };
        }
        if (item.id === payload.id && item.amount - 1 === 0) {
          return { ...item, amount: 0, selected: false };
        }

        return item;
      });

      return { ...state, items: removeAmountToItem };
    }
    default:
      return state;
  }
};

export const useCaneloneTradicional = () => {
  const [caneloneTradicional_state, dispatch] = useReducer(
    menuCaneloneTradicional,
    menu_caneloneTradicional,
  );

  return {
    data: { caneloneTradicional_state },
    functions: { dispatchCaneloneTradicional: dispatch },
  };
};
