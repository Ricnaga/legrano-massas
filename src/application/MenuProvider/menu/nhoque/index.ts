import { useReducer } from 'react';
import { UseMenuProviderData } from '../../hooks/useMenuProvider';
import { menu_nhoque } from './items';

export enum NhoqueActionsType {
  ADD_TO_CART = 'ADD_TO_CART',
  ADD_AMOUNT = 'ADD_AMOUNT',
  REMOVE_FROM_CART = 'REMOVE_FROM_CART',
}

type NhoqueState = UseMenuProviderData;

type NhoqueActions = {
  action: NhoqueActionsType;
  payload: {
    id: string;
  };
};

const menuNhoque = (state: NhoqueState, { action, payload }: NhoqueActions) => {
  switch (action) {
    case NhoqueActionsType.ADD_TO_CART: {
      const { items } = state;

      const selectItem = items.map((item) =>
        item.id === payload.id ? { ...item, selected: true, amount: 1 } : item,
      );
      return { ...state, items: selectItem };
    }
    case NhoqueActionsType.ADD_AMOUNT: {
      const { items } = state;
      const addMountToItem = items.map((item) =>
        item.id === payload.id ? { ...item, amount: item.amount + 1 } : item,
      );

      return { ...state, items: addMountToItem };
    }

    case NhoqueActionsType.REMOVE_FROM_CART: {
      const { items } = state;
      const removeAmountToItem = items.map((item) => {
        if (item.id === payload.id && item.amount - 1 > 0) {
          return { ...item, amount: item.amount - 1 };
        }
        if (item.id === payload.id && item.amount - 1 === 0) {
          return { ...item, amount: 0, selected: false };
        }

        return item;
      });

      return { ...state, items: removeAmountToItem };
    }
    default:
      return state;
  }
};

export const useNhoque = () => {
  const [nhoque_state, dispatch] = useReducer(menuNhoque, menu_nhoque);

  return {
    data: { nhoque_state },
    functions: { dispatchNhoque: dispatch },
  };
};
