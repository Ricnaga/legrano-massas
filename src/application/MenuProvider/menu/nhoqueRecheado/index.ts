import { useReducer } from 'react';
import { UseMenuProviderData } from '../../hooks/useMenuProvider';
import { menu_nhoqueRecheado } from './items';

export enum NhoqueRecheadoActionsType {
  ADD_TO_CART = 'ADD_TO_CART',
  ADD_AMOUNT = 'ADD_AMOUNT',
  REMOVE_FROM_CART = 'REMOVE_FROM_CART',
}

type NhoqueRecheadoState = UseMenuProviderData;

type NhoqueRecheadoActions = {
  action: NhoqueRecheadoActionsType;
  payload: {
    id: string;
  };
};

const menuNhoqueRecheado = (
  state: NhoqueRecheadoState,
  { action, payload }: NhoqueRecheadoActions,
) => {
  switch (action) {
    case NhoqueRecheadoActionsType.ADD_TO_CART: {
      const { items } = state;

      const selectItem = items.map((item) =>
        item.id === payload.id ? { ...item, selected: true, amount: 1 } : item,
      );
      return { ...state, items: selectItem };
    }
    case NhoqueRecheadoActionsType.ADD_AMOUNT: {
      const { items } = state;
      const addMountToItem = items.map((item) =>
        item.id === payload.id ? { ...item, amount: item.amount + 1 } : item,
      );

      return { ...state, items: addMountToItem };
    }

    case NhoqueRecheadoActionsType.REMOVE_FROM_CART: {
      const { items } = state;
      const removeAmountToItem = items.map((item) => {
        if (item.id === payload.id && item.amount - 1 > 0) {
          return { ...item, amount: item.amount - 1 };
        }
        if (item.id === payload.id && item.amount - 1 === 0) {
          return { ...item, amount: 0, selected: false };
        }

        return item;
      });

      return { ...state, items: removeAmountToItem };
    }
    default:
      return state;
  }
};

export const useNhoqueRecheado = () => {
  const [nhoqueRecheado_state, dispatch] = useReducer(
    menuNhoqueRecheado,
    menu_nhoqueRecheado,
  );

  return {
    data: { nhoqueRecheado_state },
    functions: { dispatchNhoqueRecheado: dispatch },
  };
};
