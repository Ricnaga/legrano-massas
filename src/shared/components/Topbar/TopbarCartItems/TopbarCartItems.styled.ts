import { Card, css, styled } from '@mui/material';

export const StyledCard = styled(Card)(
  ({ theme }) => css`
    padding: ${theme.spacing(1)};
    margin-top: ${theme.spacing(1)};
    width: 100%;
  `,
);
