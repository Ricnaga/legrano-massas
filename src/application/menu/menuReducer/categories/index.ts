import { caneloneEspinafre, caneloneTradicional } from './caneloneTradicional';
import { lasanha } from './lasanha';
import { nhoque, nhoqueRecheado } from './nhoque';
import { ravioli, ravioliEspinafre } from './ravioli';
import { rondele, rondeleEspinafre } from './rondele';
import { talharim } from './talharim';

export const categories = [
  nhoque,
  nhoqueRecheado,
  caneloneTradicional,
  caneloneEspinafre,
  lasanha,
  ravioli,
  ravioliEspinafre,
  rondele,
  rondeleEspinafre,
  talharim,
];
