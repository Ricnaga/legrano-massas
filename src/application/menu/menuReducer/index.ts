import { useReducer } from 'react';
import { categories } from './categories';
import {
  clearCartItems,
  getCartItems,
  saveCartItems,
  updateCartItems,
} from './storage/localStorage';

export type MenuReducerData = {
  id: string;
  category: string;
  items: Array<{
    id: string;
    name: string;
    price: number;
    weight: string;
    amount: number;
    selected: boolean;
  }>;
};

export enum MenuActionsType {
  ADD_TO_CART = 'ADD_TO_CART',
  ADD_AMOUNT = 'ADD_AMOUNT',
  REMOVE_FROM_CART = 'REMOVE_FROM_CART',
  CLEAR_CART = 'CLEAR_CART',
}

export type MenuState = Array<MenuReducerData>;
type MenuActions = {
  action: MenuActionsType;
  payload: {
    categoryId: string | null;
    itemId: string | null;
  };
};

const openMenu = (state: MenuState, { action, payload }: MenuActions) => {
  switch (action) {
    case MenuActionsType.ADD_TO_CART: {
      const addedItemToCart = state.map((selectedState) =>
        selectedState.id === payload.categoryId
          ? {
              ...selectedState,
              items: selectedState.items.map((item) =>
                item.id === payload.itemId
                  ? { ...item, selected: true, amount: 1 }
                  : item,
              ),
            }
          : selectedState,
      );
      saveCartItems(addedItemToCart);
      return addedItemToCart;
    }
    case MenuActionsType.ADD_AMOUNT: {
      const addedAmountToItems = state.map((selectedState) =>
        selectedState.id === payload.categoryId
          ? {
              ...selectedState,
              items: selectedState.items.map((item) =>
                item.id === payload.itemId
                  ? { ...item, amount: item.amount + 1 }
                  : item,
              ),
            }
          : selectedState,
      );
      updateCartItems(addedAmountToItems);
      return addedAmountToItems;
    }
    case MenuActionsType.REMOVE_FROM_CART: {
      const filteredCartItems = state.map((selectedState) =>
        selectedState.id === payload.categoryId
          ? {
              ...selectedState,
              items: selectedState.items.map((item) => {
                if (item.id === payload.itemId && item.amount - 1 > 0) {
                  return { ...item, amount: item.amount - 1 };
                }

                if (item.id === payload.itemId && item.amount - 1 === 0) {
                  return { ...item, amount: 0, selected: false };
                }

                return item;
              }),
            }
          : selectedState,
      );
      updateCartItems(filteredCartItems);
      return filteredCartItems;
    }
    case MenuActionsType.CLEAR_CART:
      clearCartItems();
      return categories;
    default:
      return state;
  }
};

export const menuReducer = () => {
  const STORED_CATEGORIES = getCartItems() ?? categories;
  const [menuState, dispatch] = useReducer(openMenu, STORED_CATEGORIES);

  return {
    data: {
      menuState,
    },
    funtions: {
      dispatchMenu: dispatch,
    },
  };
};
