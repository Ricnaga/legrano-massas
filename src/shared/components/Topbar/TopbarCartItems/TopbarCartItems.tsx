import { Add as AddIcon, Remove as RemoveIcon } from '@mui/icons-material';
import { Chip, Grid, IconButton, Typography } from '@mui/material';
import { MenuReducerData } from '../../../../application/MenuProvider/menuReducer';
import { StyledCard } from './TopbarCartItems.styled';

type TopbarCartItemsProps = Record<'cartItems', Array<MenuReducerData>> & {
  addAmount: (categoryId: string, itemId: string) => void;
  removeAmount: (categoryId: string, itemId: string) => void;
};

export function TopbarCartItems({
  cartItems,
  addAmount,
  removeAmount,
}: TopbarCartItemsProps) {
  return (
    <Grid container flexDirection="column">
      {cartItems.length ? (
        <>
          {cartItems.map(({ category, id, items }) => (
            <Grid key={id}>
              <Typography
                align="center"
                variant="h3"
                color="secondary"
                marginX={8}
              >
                {category}
              </Typography>
              <Grid
                container
                alignItems="center"
                padding={1}
                justifyContent="center"
                flexDirection="column"
              >
                {items.map((item) => (
                  <StyledCard elevation={4} key={item.id}>
                    <Typography variant="body1" color="secondary">
                      {item.name}
                    </Typography>
                    <Grid
                      container
                      alignItems="center"
                      justifyContent="space-between"
                    >
                      <Grid>
                        <IconButton
                          color="secondary"
                          onClick={() => addAmount(id, item.id)}
                        >
                          <AddIcon />
                        </IconButton>
                        <Chip
                          label={
                            <Typography variant="body2">
                              x{item.amount}
                            </Typography>
                          }
                          color="primary"
                        />
                        <IconButton
                          color="secondary"
                          onClick={() => removeAmount(id, item.id)}
                        >
                          <RemoveIcon />
                        </IconButton>
                      </Grid>
                      <Typography variant="body1" color="secondary">
                        {new Intl.NumberFormat('pt-br', {
                          style: 'currency',
                          currency: 'BRL',
                        }).format(item.price * item.amount)}
                      </Typography>
                    </Grid>
                  </StyledCard>
                ))}
                <Typography
                  align="center"
                  variant="h4"
                  color="secondary"
                  marginX={8}
                >
                  Total:
                  {new Intl.NumberFormat('pt-br', {
                    style: 'currency',
                    currency: 'BRL',
                  }).format(
                    items.reduce(
                      (accumulator, element) =>
                        accumulator + element.price * element.amount,
                      0,
                    ),
                  )}
                </Typography>
              </Grid>
            </Grid>
          ))}
        </>
      ) : (
        <Typography align="center" variant="h3" color="secondary" marginX={8}>
          Carrinho vazio!
        </Typography>
      )}
    </Grid>
  );
}
