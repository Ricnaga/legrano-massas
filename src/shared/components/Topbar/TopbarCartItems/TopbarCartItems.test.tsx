import { waitFor } from '@testing-library/react';
import {
  renderWithTheme,
  screen,
  userEvent,
} from '../../../../application/test/testing-library';
import { TopbarCartItems } from './TopbarCartItems';
import { MenuReducerData } from '../../../../application/menu/menuReducer';

let cartItems: Array<MenuReducerData> = [
  {
    category: 'mocked category',
    id: String(1),
    items: [
      {
        amount: 1,
        id: String(1),
        name: 'items name',
        price: 10,
        selected: false,
        weight: '10kg',
      },
    ],
  },
];

const mockedAddAmount = () => {
  cartItems = cartItems.map((cart) => ({
    ...cart,
    items: cart.items.map((item) => ({ ...item, amount: item.amount + 1 })),
  }));
};

const mockedRemoveAmount = () => {
  cartItems = cartItems.map((cart) => ({
    ...cart,
    items: cart.items.map((item) => ({ ...item, amount: item.amount - 1 })),
  }));
};

let containerComp: HTMLElement;
let unMountComponent: () => void;

describe('Component: TopbarCartItems', () => {
  beforeEach(() => {
    const { container, unmount } = renderWithTheme(
      <TopbarCartItems
        addAmount={mockedAddAmount}
        whatsAppText=""
        cartItems={cartItems}
        clearItems={jest.fn()}
        removeAmount={mockedRemoveAmount}
        totalPrice="mocked totalPrice"
      />,
    );
    containerComp = container;
    unMountComponent = unmount;
  });

  it('should render correctly', () => {
    expect(containerComp).toMatchSnapshot();
  });

  it('should test addAmount function', async () => {
    await userEvent.click(screen.getByTestId('AddIcon'));

    waitFor(() => expect(screen.getByText(/x2/i)).toBeInTheDocument());
  });

  it('should test removeAmount function', async () => {
    await userEvent.click(screen.getByTestId('RemoveIcon'));
    waitFor(() => expect(screen.getByText(/x1/i)).toBeInTheDocument());
  });

  it('should test empty cart', async () => {
    unMountComponent();
    renderWithTheme(
      <TopbarCartItems
        addAmount={jest.fn()}
        whatsAppText=""
        cartItems={[]}
        clearItems={jest.fn()}
        removeAmount={jest.fn()}
        totalPrice="mocked totalPrice"
      />,
    );

    expect(
      screen.getByRole('heading', {
        name: /carrinho vazio!/i,
      }),
    ).toBeInTheDocument();
  });
});
