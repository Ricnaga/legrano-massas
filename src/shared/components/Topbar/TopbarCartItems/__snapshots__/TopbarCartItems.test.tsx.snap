// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TopbarCartItems should render correctly 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container css-lhc529-MuiGrid-root"
  >
    <div
      class="MuiGrid-root css-vj1n65-MuiGrid-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-1lh1l4k-MuiTypography-root"
      >
        mocked category
      </h3>
      <div
        class="MuiGrid-root MuiGrid-container css-neeqay-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiCard-root css-16j22s2-MuiPaper-root-MuiCard-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-8r1d82-MuiTypography-root"
          >
            items name
          </p>
          <div
            class="MuiGrid-root MuiGrid-container css-1oukqxz-MuiGrid-root"
          >
            <div
              class="MuiGrid-root css-vj1n65-MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium css-vucyx8-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AddIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorPrimary MuiChip-filledPrimary css-scsbqw-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-b240w0-MuiTypography-root"
                  >
                    x
                    1
                  </p>
                </span>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium css-vucyx8-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="RemoveIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13H5v-2h14v2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-8r1d82-MuiTypography-root"
            >
              R$Â 10,00
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-10uwgf5-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-ghacns-MuiGrid-root"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-1bybvmm-MuiTypography-root"
        >
          Total:
          mocked totalPrice
        </h4>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-5 css-1e830u-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedWarning MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedWarning MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-14p4eni-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CleaningServicesIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 11h-1V3c0-1.1-.9-2-2-2h-2c-1.1 0-2 .9-2 2v8H8c-2.76 0-5 2.24-5 5v7h18v-7c0-2.76-2.24-5-5-5zm3 10h-2v-3c0-.55-.45-1-1-1s-1 .45-1 1v3h-2v-3c0-.55-.45-1-1-1s-1 .45-1 1v3H9v-3c0-.55-.45-1-1-1s-1 .45-1 1v3H5v-5c0-1.65 1.35-3 3-3h8c1.65 0 3 1.35 3 3v5z"
            />
          </svg>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-10phv9b-MuiTypography-root"
          >
            Limpar
          </h6>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-5 css-1e830u-MuiGrid-root"
      >
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-2umqei-MuiButtonBase-root-MuiButton-root"
          href="http://wa.me/5511910371645?text="
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="WhatsAppIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.75 13.96c.25.13.41.2.46.3.06.11.04.61-.21 1.18-.2.56-1.24 1.1-1.7 1.12-.46.02-.47.36-2.96-.73-2.49-1.09-3.99-3.75-4.11-3.92-.12-.17-.96-1.38-.92-2.61.05-1.22.69-1.8.95-2.04.24-.26.51-.29.68-.26h.47c.15 0 .36-.06.55.45l.69 1.87c.06.13.1.28.01.44l-.27.41-.39.42c-.12.12-.26.25-.12.5.12.26.62 1.09 1.32 1.78.91.88 1.71 1.17 1.95 1.3.24.14.39.12.54-.04l.81-.94c.19-.25.35-.19.58-.11l1.67.88M12 2a10 10 0 0 1 10 10 10 10 0 0 1-10 10c-1.97 0-3.8-.57-5.35-1.55L2 22l1.55-4.65A9.969 9.969 0 0 1 2 12 10 10 0 0 1 12 2m0 2a8 8 0 0 0-8 8c0 1.72.54 3.31 1.46 4.61L4.5 19.5l2.89-.96A7.95 7.95 0 0 0 12 20a8 8 0 0 0 8-8 8 8 0 0 0-8-8z"
            />
          </svg>
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 css-10phv9b-MuiTypography-root"
          >
            Me manda no zap
          </h6>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
